met-web/src/components/engagement/form/ActionContext.tsx

```diff
@@ -101,7 +100,6 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
 
     // Update states based on the loaded engagement.
     const setEngagement = (engagement: Engagement) => {
-        console.log('setEngagement', engagement);
```

Comment:
The useState "setEngagement" probably won't have the new value here yet since those assignments are asynchronous. I know you are returning the engagement, but the console.log could be misleading.

Reply:
Good thing I'm removing it in this commit :P